### DISCUSSION TITLE
Protect my swagger ui docs and API with authentication

### DISCUSSION THREAD

**Discussion by jonathanrsmjtk** [fastapi/fastapi#8192] [Questions] [answered]
**Description**

is it possible to protect my API and UI docs using an authentication process? so only some user can access my API, not others due to my secret data

**Comment by euri10** [comment, marked as answer]
I think this will help https://github.com/tiangolo/fastapi/issues/364


> The solution discussed in the article is to disabled the automatic generated endpoints and create your own with authentication

https://github.com/fastapi/fastapi/issues/364#issuecomment-760949372

**Comment by jonathanrsmjtk** [comment]
> I think this will help #364

thanks! it works!

**Comment by tiangolo** [comment]
Thanks for the help here @euri10 ! :clap: :bow: 

Thanks for reporting back and closing the issue @jonathanrsmjtk  :+1:
